src/components/Clock/Clock.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import { Button } from '@material-ui/core';
+
+import { Box } from '@material-ui/core';
+import IconButton from '@material-ui/core/IconButton';
+import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
+import RemoveCircleOutlineIcon from '@material-ui/icons/RemoveCircleOutline';
+import { ClockBox, TimeBox } from './styles';
+import { ClockTimer } from './types';
+
+const Clock: React.FC = () => {
+    const [ clockObject, setClockObject ] = React.useState({   
+        minutes: 0,
+        seconds: 0,
+    });
+    const formatTime = (time : number) : string =>
+    {
+        
+        // const formattedClockObject = clockObject;
+        // console.log(formattedClockObject);
+        // if (Number(formattedClockObject.minutes) < 10 && clockObject.minutes.length < 2) 
+        // {
+        //     formattedClockObject.minutes = `0${formattedClockObject.minutes}`;
+        // }
+        // if (Number(formattedClockObject.seconds) < 10 && formattedClockObject.seconds.length < 2) 
+        // {
+        //     formattedClockObject.seconds = `0${formattedClockObject.seconds}`;
+        // }
+    
+        //console.log(formattedClockObject);
+        //setClockObject(formattedClockObject);
+       
+        return `${String(time).length < 2 ? '0' : ''}${time}`;
+    };
+
+    function increment(time : number) {
+ 
+        clockObject.seconds = clockObject.seconds + time;
```

Comment:
Avoid changing a state like this. All the state changes should be handled by the function returned from `useState` - in this case, `setClockObject`. Mutating state like this might cause unexpected results